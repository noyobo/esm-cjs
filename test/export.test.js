const transform = require('../lib/transform');

describe('transform export', () => {
  let code;
  it('should exporting individual features', () => {
    code = transform(`export let a, b, c;`);
    expect(code).toMatchSnapshot();
    code = transform(`export let a = 1;`);
    expect(code).toMatchSnapshot();
    code = transform(`export let a = 1, b = 2, c = 3;`);
    expect(code).toMatchSnapshot();
    code = transform(`export function foo(){ /**/ };`);
    expect(code).toMatchSnapshot();
    code = transform(`export class ClassName { }`);
    expect(code).toMatchSnapshot();
  });

  it('should export list', () => {
    const code = transform(`const a = 1; const b = 2; export { a, b }`);
    expect(code).toMatchSnapshot();
  });

  it('should renaming exports', () => {
    const code = transform(`var variable1 = 1; var variable2 = 2; export { variable1 as name1, variable2 as name2 };`);
    expect(code).toMatchSnapshot();
  });

  it('should default exports', () => {
    code = transform(`export default true`);
    expect(code).toMatchSnapshot();
    code = transform(`export default function () {  }`);
    expect(code).toMatchSnapshot();
    code = transform(`export default function name1() {  }`);
    expect(code).toMatchSnapshot();
    code = transform(`export default class ClassName {}`);
    expect(code).toMatchSnapshot();
  });

  it('should aggregating modules', () => {
    const fn = jest.fn(() => {
      transform(`export { name1, name2 } from 'foo'`);
    });
    expect(fn).toThrowErrorMatchingSnapshot();
  });
});
