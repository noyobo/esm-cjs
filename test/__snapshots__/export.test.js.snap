// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform export should aggregating modules 1`] = `"Not support export aggregating modules"`;

exports[`transform export should default exports 1`] = `
"const _default = true;
exports.default = _default;"
`;

exports[`transform export should default exports 2`] = `
"function _default() {}

exports.default = _default;"
`;

exports[`transform export should default exports 3`] = `
"function name1() {}

exports.default = name1;"
`;

exports[`transform export should default exports 4`] = `
"class ClassName {}

exports.default = ClassName;"
`;

exports[`transform export should export list 1`] = `
"const a = 1;
const b = 2;
exports.a = a;
exports.b = b;"
`;

exports[`transform export should exporting individual features 1`] = `
"let a, b, c;
exports.a = a;
exports.b = b;
exports.c = c;"
`;

exports[`transform export should exporting individual features 2`] = `
"let a = 1;
exports.a = a;"
`;

exports[`transform export should exporting individual features 3`] = `
"let a = 1,
    b = 2,
    c = 3;
exports.a = a;
exports.b = b;
exports.c = c;"
`;

exports[`transform export should exporting individual features 4`] = `
"function foo() {
  /**/
}

exports.foo = foo;
;"
`;

exports[`transform export should exporting individual features 5`] = `
"class ClassName {}

exports.ClassName = ClassName;"
`;

exports[`transform export should renaming exports 1`] = `
"var variable1 = 1;
var variable2 = 2;
exports.name1 = variable1;
exports.name2 = variable2;"
`;
